tests.py

```diff
@@ -14,6 +14,27 @@
 
 
 if __name__ == '__main__':
+
+    try:
+        import setuptools
+    except ImportError:
+        pass
+
+    # The warnings need to be before any of matplotlib imports, but after
+    # setuptools (if present) which has syntax error with the warnings enabled.
+    # Filtering by module does not work as this will be raise by Python itself.
+    # so `module=matplotlib.*` is out of questions.
+
+    import warnings
+
+    # Python 3.6 deprecate invalid character-pairs \A, \* ... in non raw-strings
+    # and other things. Let's not re-introduce them
+    warnings.filterwarnings('error', category=DeprecationWarning)
```

Comment:
Though I prefer not to introduce any deprecations either, this seems a bit broad for this PR.